src/components/atom/Modal/index.tsx

```diff
@@ -0,0 +1,46 @@
+import { FC, ReactNode, useEffect } from "react";
+import { ReactComponent as ModalCloseIcon } from "../../../assets/img/modal-close-icon.svg";
+
+interface ModalProps {
+  isOpen: boolean;
+  modalHeaderText?: string;
+  onClose: () => void;
+  children: ReactNode;
+}
+
+const Modal: FC<ModalProps> = ({ isOpen, onClose, children, modalHeaderText }) => {
+  useEffect(() => {
+    const handleEscape = (event: KeyboardEvent) => {
+      if (event.key === "Escape") {
+        onClose();
+      }
+    };
+
+    if (isOpen) {
+      window.addEventListener("keydown", handleEscape);
+    }
+
+    return () => {
+      window.removeEventListener("keydown", handleEscape);
+    };
+  }, [isOpen, onClose]);
+
+  if (!isOpen) return null;
+
+  return (
+    <div className="fixed inset-0 z-50 flex items-center justify-center">
```

Comment:
What happens if user click outside modal? Expected behavior is to close modal.

Reply:
It has {onClose} function on background layer of modal. it closes on outside click

Reply:
@npanajotov Can we use the tailwind components that we have for modals instead of making it from scratch? We have that